components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/store/JDBCRecoveryDataStore.java

```diff
@@ -244,6 +370,137 @@ public UserRecoveryData load(String code, boolean skipExpiryValidation) throws I
         throw Utils.handleClientException(ERROR_CODE_INVALID_CODE, code);
     }
 
+    @Override
+    public UserRecoveryData loadFromRecoveryFlowId(String recoveryFlowId, Enum recoveryStep)
+            throws IdentityRecoveryException {
+
+        handleRecoveryDataEventPublishing(PRE_GET_USER_RECOVERY_DATA,
+                GET_USER_RECOVERY_DATA_SCENARIO_WITH_CODE_EXPIRY_VALIDATION, null, null, null, null,
+                new UserRecoveryData(null, recoveryFlowId, null, null, recoveryStep));
+
+        PreparedStatement prepStmt1 = null;
+        PreparedStatement prepStmt2 = null;
+        ResultSet resultSet1 = null;
+        ResultSet resultSet2 = null;
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+
+        User user = null;
+        String code = null;
+        int attempt = 0;
+        int resendCount =0;
+        long createdTimeStamp = 0;
+        UserRecoveryData userRecoveryData = null;
+        Boolean isOperationSuccess = false;
```

Comment:
any reason not to use the primitive type?

Reply:
This variable is passed as a parameter to the existing method "handleRecoveryDataEventPublishing" [1], where it needs a parameter with `Boolean` type.

[1] https://github.com/wso2-extensions/identity-governance/blob/1f7ac6671dc03afe2dfe51196c89cfa8ff15b22f/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/store/JDBCRecoveryDataStore.java#L1133